packages/sdk/src/swap/v2/types.ts

```diff
@@ -47,27 +47,39 @@ interface DepositFields {
   failedBlockIndex: string | undefined;
 }
 
-interface SwapChunk {
-  swapInputAmount: string;
-  swapOutputAmount: string | undefined;
-  scheduledAt: number;
-  scheduledBlockIndex: string;
-  executedAt: number | undefined;
-  executedBlockIndex: string | undefined;
-  retryCount: number;
-  latestSwapRescheduledAt: number | undefined;
-  latestSwapRescheduledBlockIndex: number | undefined;
-  fees: PaidFee[];
-  isDca: false;
-}
-
 interface SwapFields {
-  totalInputAmountSwapped: string | undefined;
-  totalOutputAmountSwapped: string | undefined;
-  totalChunksExecuted: number;
-  isDca: true;
-  lastExecutedChunk: SwapChunk | undefined;
-  currentChunk: SwapChunk;
+  originalInputAmount: string;
+  remainingInputAmount: string;
+  swappedInputAmount: string;
+  swappedOutputAmount: string;
+  regular?: {
+    inputAmount: string;
+    outputAmount: string;
+    scheduledAt: number;
+    scheduledBlockIndex: string;
+    executedAt?: number;
+    executedBlockIndex?: string;
+    retryCount: 0;
+  };
+  dca?: {
+    lastExecutedChunk: {
+      inputAmount: string;
+      outputAmount: string;
+      scheduledAt: number;
+      scheduledBlockIndex: string;
+      executedAt: number;
+      executedBlockIndex: string;
+      retryCount: number;
+    } | null;
+    currentChunk: {
+      inputAmount: string;
+      scheduledAt: number;
+      scheduledBlockIndex: string;
+      retryCount: number;
+    } | null;
```

Comment:
not sure if this can be null

Reply:
it can be null if the entire DCA swap is completed